client/src/bundles/ConfirmFolder.js

```diff
@@ -0,0 +1,148 @@
+/* global window */
+import i18n from 'i18n';
+import jQuery from 'jquery';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { loadComponent } from 'lib/Injector';
+
+const FormBuilderModal = loadComponent('FormBuilderModal');
+
+jQuery.entwine('ss', ($) => {
+  /** Mark newly added fields as new */
+  $('#Form_EditForm_Fields').entwine({
+    onmatch() {
+      this._super();
+
+      this.on('addnewinline', () => {
+        this.one('reload', () => {
+          const newField = this.find('.ss-gridfield-item').last();
+          newField.attr('data-new', 1);
+        });
+      });
+    }
+  });
+
+  /** Move our options under the radio button areas */
+  $('#Form_Form_FolderOptions-new').entwine({
+    onmatch() {
+      $('#Form_Form_CreateFolder_Holder').detach().appendTo($('#Form_Form_FolderOptions-new').parent().parent());
+
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'none' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+    },
+    onchange() {
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'none' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+    }
+  });
+
+  /** Move our options under the radio button areas */
+  $('#Form_Form_FolderOptions-existing').entwine({
+    onmatch() {
+      $('#Form_Form_FolderID_Holder').detach().appendTo($('#Form_Form_FolderOptions-existing').parent().parent());
+
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'none' : 'block');
+    },
+    onchange() {
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'none' : 'block');
+    }
+  });
```

Comment:
😭 This is what I was worried about with the FormSchema approach - there's no way to represent the UI in schema, so we end up having to shuffle DOM around afterwards.

I don't know that there's a change worth making to this, but we should definitely look into developing the ability to nest fields from the PHP API at some point.